core/src/main/scala/cats/syntax/applicativeError.scala

```diff
@@ -83,6 +85,9 @@ final class ApplicativeErrorOps[F[_], E, A](private val fa: F[A]) extends AnyVal
   def attempt(implicit F: ApplicativeError[F, E]): F[Either[E, A]] =
     F.attempt(fa)
 
+  def attemptNarrow[EE](implicit F: ApplicativeError[F, E], tag: ClassTag[EE], ev: EE <:< E): F[Either[EE, A]] =
```

Comment:
why do we need a classtag here?

Reply:
`EE` would be erased and we wouldn't be able to pattern match on it otherwise @tkroman

Reply:
It's unsound without `ev` due to erasure, but `ev` is unused in the body so we might want to add a comment to this effect.